src/main.rs

```diff
@@ -90,14 +90,23 @@ fn print_page(
         let reader = BufReader::new(file);
 
         if enable_markdown {
+            let mut stdout = std::io::stdout();
+            stdout.lock();
```

Comment:
I am surprised the compiler / linter didn't catch this: This returns a lock that can be written to, it doesn't actually lock stdout for our variable `stdout`. So this should roughly be `let stdout = io::stdout().lock()`

You also do this in both branches of this conditional, so it can be moved out :)